src/index.js

```diff
@@ -16,136 +16,146 @@ const { getAccount, getLastOrder } = require('./core/account');
 // product related functions
 const { getSnapshot } = require('./core/product');
 
-function reactivate(time) {
-    setInterval(attemptRun, time);
+function reactivate(time, coin) {
+    setInterval(attemptRun(coin), time);
     logIt({
         title: 'checking again',
         info: moment().add(time, 'milliseconds').fromNow(),
     });
 }
 
 function attemptRun() {
-    try {
-        run();
-    } catch (e) {
-        logIt({
-            form: 'error',
-            title: 'failed to run',
-            info: e,
-        });
-        reactivate(FIFTEEN_MINS_MS);
-    }
+    require('dotenv').config();
+    const currency = process.env.CURRENCY;
+    coins.map(coin => {
```

Comment:
@DanielMSchmidt this is where things need to be async. I think I have to transform the entire guts of run to something promise based?